src/Common/CodeAnalysis.ruleset

```diff
@@ -0,0 +1,38 @@
+﻿<?xml version="1.0"?>
+<RuleSet Name="Code analysis rules for the Q# compiler" ToolsVersion="14.0">
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers.ReadabilityRules">
+    <Rule Id="SA1122" Action="None" />      <!-- Use string.Empty for empty strings -->
+    <Rule Id="SA1127" Action="None" />      <!-- Generic type constraints should be on their own line -->
+  </Rules>
+
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers.OrderingRules">
+    <Rule Id="SA1200" Action="None" />      <!-- Using directives should be placed correctly -->
+    <Rule Id="SA1201" Action="None" />      <!-- Elements should appear in the correct order -->
+    <Rule Id="SA1202" Action="None" />      <!-- Elements should be ordered by access -->
+    <Rule Id="SA1203" Action="None" />      <!-- Constants should appear before fields -->
+  </Rules>
+
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers.MaintainabilityRules">
+    <Rule Id="SA1401" Action="None" />      <!-- Fields should be private -->
+    <Rule Id="SA1402" Action="None" />      <!-- File may only contain a single type -->
+    <Rule Id="SA1405" Action="None" />      <!-- Debug.Assert should provide message text -->
+    <Rule Id="SA1413" Action="None" />      <!-- Use trailing comma in multi-line initializers -->
+  </Rules>
+
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers.LayoutRules">
+    <Rule Id="SA1503" Action="None" />      <!-- Braces should not be omitted -->
+    <Rule Id="SA1512" Action="None" />      <!-- Single-line comments should not be followed by blank line -->
+    <Rule Id="SA1513" Action="None" />      <!-- Closing brace should be followed by blank line -->
+  </Rules>
+
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers.DocumentationRules">
+    <Rule Id="SA1600" Action="None" />      <!-- Elements should be documented -->
+    <Rule Id="SA1602" Action="None" />      <!-- Enumeration items should be documented -->
+    <Rule Id="SA1611" Action="None" />      <!-- Element parameters should be documented -->
+    <Rule Id="SA1615" Action="None" />      <!-- Element return value should be documented -->
```

Comment:
I'm not in favor of these two rules. Documenting every parameter and every return value is tedious and often unhelpful as documentation. We haven't been documenting along these rules so far, so I don't think we should introduce these now, in this PR.

Reply:
These rules are disabled with `Action="None"`. :) Sorry, this file is pretty confusing, because the rule descriptions are phrased like they are being enabled, but all the rules in this file are being disabled - they are normally on by default in StyleCop.